src/widgets/createTable.tsx

```diff
@@ -0,0 +1,44 @@
+import {
+	useReactTable,
+	getCoreRowModel,
+	getPaginationRowModel,
+	getSortedRowModel,
+	type SortingState,
+	type AccessorKeyColumnDefBase,
+	type IdIdentifier,
+} from "@tanstack/react-table";
+import { useState } from "react";
+
+export default function createTable<T>({
```

Comment:
I think this function is a hook, since it uses other hooks (`useState`, `useReactTable`) under the hood, and [its name should therefore begin with `use`](https://react.dev/learn/reusing-logic-with-custom-hooks#hook-names-always-start-with-use).